test/skate_web/channels/alerts_channel_test.exs

```diff
@@ -24,6 +24,18 @@ defmodule SkateWeb.AlertsChannelTest do
       assert {:ok, %{data: ["Some alert"]}, _socket} =
                subscribe_and_join(socket, AlertsChannel, "alerts:route:" <> route_id)
     end
+
+    test "returns an error when trying to join with expired token", %{socket: socket} do
+      reassign_env(:skate, :valid_token_fn, fn _socket -> false end)
+
+      for %{result: result, route: route} <- [
+            %{result: {:error, :not_authenticated}, route: "alerts:route:"},
+            %{result: {:error, :not_authenticated}, route: "vehicles:route:"},
+            %{result: {:error, :not_authenticated}, route: "random:topic:1"},
+            %{result: {:error, :not_authenticated}, route: "random:topic:2"}
+          ],
+          do: assert(^result = subscribe_and_join(socket, AlertsChannel, route))
+    end
```

Comment:
**suggestion:** it's a little confusing to have this specifically in the `AlertsChannel` test since it tests several different channels. One option is to simply make this a bunch of different tests in different channel modules, or maybe there's some good unified place to put this one test that makes sense.

Reply:
Oh that reminds me I need to go over and change all the test names, since I just copy pasted.

Reply:
also this is only testing `AlertsChannel`, I was leaving random topics in there when testing to ensure I was hitting strange edge cases should they appear.